File: warehouse-server/schemas/book.js
'use strict'

var Mongoose = require('mongoose'),
    schema = new Mongoose.Schema({
        isbn: {
            type: String,
            required: true,
            unique: true
        },
        title: {
            type: String,
            required: true
        },
        author: {
            type: String,
            required: true
        },
        year: {
            type: Number,
            required: true
        },
        publisher: {
            type: String,
            required: true
        },
        image: {
            type: String
        },
        price: {
            type: Number
        },
        stock: {
            type: Number,
            required: true
        }
    });

var item = Mongoose.model('Book', schema);

module.exports = schema;

File: win-clients/warehouse-client/Objects/Book.cs
namespace warehouse_client.Objects
{
    public class Book
    {
        public Book()
        {

        }

        public string isbn { get; set; }
        public string title { get; set; }
        public string author { get; set; }
        public int year { get; set; }
        public string publisher { get; set; }
        public float price { get; set; }
        public int stock { get; set; }
        public string image { get; set; }
    }
}